src/urllib3/_collections.py

```diff
@@ -58,7 +58,6 @@ def __getitem__(self, key):
             return item
 
     def __setitem__(self, key, value):
-        evicted_value = _Null
```

Comment:
I actually don't agree that this is an improvement. Yes we "know" what `RLock.__enter__()` won't fail but it makes sense to define the `evicted_value` outside the block since it's used outside the block after it closes.